File: modules.py
from Crypto.Cipher import PKCS1_OAEP
from Crypto.PublicKey import RSA
import requests
import base64

# Reading the public key
with open("publicKey.txt", "rb") as pb:
    pubKeyObj = RSA.importKey(pb.read())

# Reading the private key
with open("privateKey.txt", "rb") as pb:
    privKeyObj = RSA.importKey(pb.read())

# takes message to be encrypted, RSA public key object -> returns encrypted binary    
def custom_encrypt(msg,pub_key_obj):
    cipher = PKCS1_OAEP.new(pub_key_obj)
    emsg = cipher.encrypt(msg.encode())
    return emsg

# takes encrypted message to be decrypted, RSA private key object -> returns decrypted string
def custom_decrypt(emsg,priv_key_obj):
    cipher = PKCS1_OAEP.new(priv_key_obj)
    plainText = cipher.decrypt(emsg).decode()
    return plainText

# Sample encryption
msg = '{"key1":"value1","key2":"value2"}'
emsg = custom_encrypt(msg,pubKeyObj)

#sample decryption
plainText = custom_decrypt(emsg,privKeyObj)

print("Encrypted: ",emsg)
print("Decrypted: ",plainText)

#sample post request and response handling
data_ = base64.b64encode(emsg)
headers = {'Content-Type': 'application/octet-stream'}
url_ = 'http://localhost:3000/upload'

res = requests.post(url=url_, data=emsg, headers=headers)
print("Status code:",res.status_code)
if(res.status_code == 200):
    print("*******Success******")
print("Response content: ",res.content)
decrypted_message = custom_decrypt(res.content,privKeyObj)
print("Response Content Decrypted: ",decrypted_message)

File: modules.ts
const express = require('express');
const cors =  require('cors');
const app = express();
const NodeRSA = require('node-rsa');
const Crypto_ = require('crypto');
var fs = require("fs");
app.use(cors());
var bodyParser = require('body-parser');

// Global Key def
const key = new NodeRSA({ b: 1024 });
key.importKey
const fsPri = fs.readFileSync("./privateKey.txt", 'utf8');
const fsPub = fs.readFileSync("./publicKey.txt", 'utf8');

// Module Functions

// Decrypt function: Takes encrypted binary, private key string -> returns decrypted string
function custom_Decrypt(emsg:any, Priv_key:any):string{
    const Priv_key_obj = new NodeRSA(Priv_key);
    const dmsg = Priv_key_obj.decrypt(emsg, 'utf8');
    return dmsg;
}

// Encrypt function: Takes string msg, public key string -> returns encrypted binary
function custom_Encrypt(msg:string, Pub_key:any):any{
    const encryptedMessage = Crypto_.publicEncrypt(Pub_key, Buffer.from(msg));
    return encryptedMessage;
}



app.post('/check', express.json(), async (req:any, res:any) => {
    var data = req.body;
    console.log('test',data);
    console.log('test',data.key1);
    console.log('test',data.key2);
    res.send('JSON uploaded successfully!');
});

app.post('/upload',bodyParser.raw({type: 'application/octet-stream', limit : '2mb'}), async (req:any, res:any) => {

    // Test Decryption
    var contents = req.body;
    console.log(contents);

    const decryptedMsg = custom_Decrypt(contents, fsPri);

    console.log(decryptedMsg)

    const test_ = JSON.parse(decryptedMsg);
    console.log(test_);
    console.log(test_.key1);
    console.log(test_.key2);

    

    const data = "AuthenticationKey==293jdq17hp9f";
    const encryptedMessage = custom_Encrypt(data,fsPub);
    console.log(encryptedMessage);
    res.send(encryptedMessage);
});



app.listen(3000, () => {
    console.log('Server running on port 3000');
});